lib/core/src/sdk.rs

```diff
@@ -1099,35 +1252,36 @@ impl LiquidSdk {
                     None => {
                         debug!("Timeout occurred without payment, set swap to timed out");
                         match swap {
-                            Swap::Send(_) => self.send_swap_state_handler.update_swap_info(&swap_id, TimedOut, None, None, None).await?,
-                            Swap::Chain(_) => self.chain_swap_state_handler.update_swap_info(&swap_id, TimedOut, None, None, None, None).await?,
+                            Swap::Send(_) => self.send_swap_state_handler.update_swap_info(&expected_swap_id, TimedOut, None, None, None).await?,
+                            Swap::Chain(_) => self.chain_swap_state_handler.update_swap_info(&expected_swap_id, TimedOut, None, None, None, None).await?,
                             _ => ()
                         }
                         return Err(PaymentError::PaymentTimeout)
                     },
                 },
                 event = events_stream.recv() => match event {
-                    Ok(SdkEvent::PaymentPending { details }) => match details.swap_id.clone() {
-                        Some(id) if id == swap_id => match accept_zero_conf {
-                            true => {
-                                debug!("Received Send Payment pending event with zero-conf accepted");
-                                return Ok(details)
+                    Ok(SdkEvent::PaymentPending { details: payment }) =>
```

Comment:
How about rewriting this branch as
```rust
Ok(SdkEvent::PaymentPending { details: payment }) => {
    let maybe_payment_swap_id = payment.details.as_ref().and_then(|d|d.get_swap_id());
    if matches!(maybe_payment_swap_id, Some(swap_id) if swap_id == expected_swap_id) {
        match accept_zero_conf {
            true => {
                debug!("Received Send Payment pending event with zero-conf accepted");
                return Ok(payment)
            }
            false => {
                debug!("Received Send Payment pending event, waiting for confirmation");
                maybe_payment = Some(payment);
            }
        }
    };
}
```

Less nested blocks (so less indentation) and IMO simpler to read.

Reply:
Agreed, it does look simpler to read